models/babyswap/bnb/babyswap_bnb_trades.sql

```diff
@@ -0,0 +1,103 @@
+{{ config(
+    alias = 'trades'
+    ,partition_by = ['block_date']
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,incremental_strategy = 'merge'
+    ,unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address']
+    ,post_hook='{{ expose_spells(\'["bnb"]\',
+                                      "project",
+                                      "babyswap",
+                                    \'["codingsh"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2021-06-01' %}
+
+WITH babyswap_dex AS (
+    SELECT  t.evt_block_time                                             AS block_time,
+            `to`                                                         AS taker,
+            sender                                                       AS maker,
+            CASE WHEN amount0Out = 0 THEN amount1Out ELSE amount0Out END AS token_bought_amount_raw,
+            CASE WHEN amount0In = 0 THEN amount1In ELSE amount0In END    AS token_sold_amount_raw,
+            cast(NULL as double)                                         AS amount_usd,
+            CASE WHEN amount0Out = 0 THEN token1 ELSE token0 END         AS token_bought_address,
+            CASE WHEN amount0In = 0 THEN token1 ELSE token0 END          AS token_sold_address,
+            t.contract_address                                           AS project_contract_address,
+            t.evt_tx_hash                                                AS tx_hash,
+            ''                                                           AS trace_address,
+            t.evt_index
+    FROM {{ source('babyswap_bnb', 'Pair_evt_Swap') }} t
+    INNER JOIN {{ source('babyswap_bnb', 'BabySwapFactory_evt_PairCreated') }} p
```

Comment:
Hi @developerfred , please verify if these sources available on dune V2 engines as well. We may need to find the decoded tables as source first then we can develop the spell/model.

Reply:
done